packages/patrol/lib/src/binding.dart

```diff
@@ -93,6 +95,29 @@ class PatrolBinding extends LiveTestWidgetsFlutterBinding {
       final nameOfRequestedTest = await patrolAppService.testExecutionRequested;
 
       if (nameOfRequestedTest == _currentDartTest) {
+        if (const bool.fromEnvironment('COVERAGE_ENABLED')) {
+          postEvent(
+            'waitForCoverageCollection',
+            {'mainIsolateId': Service.getIsolateId(Isolate.current)},
+          );
+
+          var coverageCollected = false;
+
+          registerExtension(
+            'ext.patrol.markTestCompleted',
+            (method, parameters) async {
+              coverageCollected = true;
+              return ServiceExtensionResponse.result(jsonEncode({}));
+            },
+          );
+
+          while (!coverageCollected) {
+            // The loop is needed to keep this isolate alive until the coverage
+            // data is collected.
+            await Future<void>.delayed(const Duration(seconds: 1));
+          }
```

Comment:
Will the Isolate be terminated while waiting for `Completer`?

Reply:
I'm not sure I understand the question. Could you be more specific?

Reply:
I mean, instead of doing this loop you could use `Completer` to wait for `ext.patrol.markTestCompleted` to finish:
```dart
  final completer = Completer<void>();

  registerExtension(
    'ext.patrol.markTestCompleted',
    (method, parameters) async {
      completer.complete();
      return ServiceExtensionResponse.result(jsonEncode({}));
    },
  );

  // Wait until the coverage is collected
  await completer.future;
  ```
  But I'm not sure if the Isolate will be terminated while waiting here `await completer.future`

Reply:
🙈 it should work just fine